browser/extensions/api/brave_rewards_api.cc

```diff
@@ -1189,6 +1195,57 @@ BraveRewardsShouldShowOnboardingFunction::Run() {
   return RespondNow(OneArgument(base::Value(should_show)));
 }
 
+BraveRewardsGetScheduledCaptchaInfoFunction::
+    ~BraveRewardsGetScheduledCaptchaInfoFunction() = default;
+
+ExtensionFunction::ResponseAction
+BraveRewardsGetScheduledCaptchaInfoFunction::Run() {
+#if BUILDFLAG(BRAVE_ADAPTIVE_CAPTCHA_ENABLED)
+  Profile* profile = Profile::FromBrowserContext(browser_context());
+  auto* brave_adaptive_captcha_service =
+      brave_adaptive_captcha::BraveAdaptiveCaptchaServiceFactory::GetForProfile(
+          profile);
+  if (!brave_adaptive_captcha_service) {
+    return RespondNow(Error("Adaptive captcha service is not initialized"));
+  }
+
+  std::string url;
+  bool max_attempts_exceeded = false;
+  brave_adaptive_captcha_service->GetScheduledCaptchaInfo(
+      &url, &max_attempts_exceeded);
+#endif
+
+  base::DictionaryValue dict;
+  dict.SetString("url", url);
+  dict.SetBoolean("maxAttemptsExceeded", max_attempts_exceeded);
+
+  return RespondNow(OneArgument(std::move(dict)));
+}
+
+BraveRewardsUpdateScheduledCaptchaResultFunction::
+    ~BraveRewardsUpdateScheduledCaptchaResultFunction() = default;
+
+ExtensionFunction::ResponseAction
+BraveRewardsUpdateScheduledCaptchaResultFunction::Run() {
+  auto params(
+      brave_rewards::UpdateScheduledCaptchaResult::Params::Create(*args_));
+  EXTENSION_FUNCTION_VALIDATE(params.get());
+
+#if BUILDFLAG(BRAVE_ADAPTIVE_CAPTCHA_ENABLED)
+  Profile* profile = Profile::FromBrowserContext(browser_context());
+  auto* brave_adaptive_captcha_service =
+      brave_adaptive_captcha::BraveAdaptiveCaptchaServiceFactory::GetForProfile(
+          profile);
+  if (!brave_adaptive_captcha_service) {
+    return RespondNow(Error("Adaptive captcha service is not initialized"));
+  }
+
+  brave_adaptive_captcha_service->UpdateScheduledCaptchaResult(params->result);
+#endif
+
+  return RespondNow(NoArguments());
```

Comment:
in this and other methods it should respond with an error if BRAVE_ADAPTIVE_CAPTCHA_ENABLED == false